generate_route.py

```diff
@@ -46,30 +40,36 @@ class Location(object):
       longitude A float for the longitude of the location in Decimal Degrees
   """
 
-  def __init__(self,latitude,longitude):
+  def __init__(self,latitude,longitude,altitude=None):
     self.latitude = latitude
     self.longitude = longitude
-    self.altitude = None
+    self.altitude = altitude
     self.x = None
     self.y = None
     self.z = None
 
   def get_lat_lon_tuple(self):
     return (self.latitude,self.longitude)
 
-  def get_XYZ(Self):
-    pass
+  def get_xyz_tuple(self):
+    return (self.x,self.y,self.z)
+
+  def add_XYZ(self):
```

Comment:
I think the issue I ran into here is that it's faster to make a batch request for the elevations of all of the locations at once instead of when each Location is initialized, and the xyz can't be calculated until there's an elevation value, so I was waiting until each point had an altitude to add the xyz

Reply:
In that case, does initializing all the Locations with altitude and XYZ's inside create_route? This is to avoid returning objects which are partially initialized.